### PR TITLE
chore(doc): add `trigger` related configuration to webhook notification examples

### PR DESCRIPTION
**Pull request by cuichenli**
The document on webhook notification is unclear on the `trigger` part. In the example of `configuration` part, there is one section for `trigger` but it is not explicitly mentioned in the document, the template section is also missing in the examples. This PR adds one extra step in configuration on `trigger` and also add the `trigger` sections in the examples. 

image

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: docs/operator-manual/notifications/services/webhook.md [hunk 1]
```diff
@@ -31,19 +31,29 @@ kind: ConfigMap
...
+  template.<template-name>: |
     webhook:
       <webhook-name>:
         method: POST # one of: GET, POST, PUT, PATCH. Default value: GET 
         path: <optional-path-template>
         body: |
           <optional-body-template>
+```
+
+3 Define trigger that customize when the request should be send and which template should be sued.
```
Comment (ashutosh16) [inline docs/operator-manual/notifications/services/webhook.md:43]: sued> used

File: docs/operator-manual/notifications/services/webhook.md [hunk 2]
```diff
@@ -99,6 +94,9 @@ data:
             "target_url": "{{.context.argocdUrl}}/applications/{{.app.metadata.name}}",
             "context": "continuous-delivery/{{.app.metadata.name}}"
           }
+  trigger.on-sync-succeeded: |
+    - when: app.status.operationState.phase in ['Succeeded']
+      send: [github-commit-status]
```
Comment (ashutosh16) [inline docs/operator-manual/notifications/services/webhook.md:99]: Should this be `send: [<template-name>]`

File: docs/operator-manual/notifications/services/webhook.md (hunk 1 above)
Comment (cuichenli) [reply docs/operator-manual/notifications/services/webhook.md]: updated

File: docs/operator-manual/notifications/services/webhook.md [hunk 3]
```diff
@@ -31,19 +31,29 @@ kind: ConfigMap
...
+  template.<template-name>: |
     webhook:
       <webhook-name>:
         method: POST # one of: GET, POST, PUT, PATCH. Default value: GET 
         path: <optional-path-template>
         body: |
           <optional-body-template>
+```
+
+3 Define trigger that customize when the request should be send and which template should be used.
```
Comment (nitishfy) [inline docs/operator-manual/notifications/services/webhook.md:43]:
```suggestion
3 Define trigger that customize when the request should be sent and which template should be used.
```

**Comment by pasha-codefresh** [comment]
This should be done inside https://github.com/argoproj/notifications-engine/blob/master/docs/services/webhook.md

@cuichenli i am sorry for late review, please reopen PR in notification engine

**Pull request closed without merging**